code/modules/antagonists/changeling/powers/adrenaline.dm

```diff
@@ -1,19 +1,53 @@
 /datum/action/changeling/adrenaline
-	name = "Adrenaline Sacs"
-	desc = "We evolve additional sacs of adrenaline throughout our body. Costs 30 chemicals."
-	helptext = "Removes all stuns instantly and adds a short-term reduction in further stuns. Can be used while unconscious. Continued use poisons the body."
+	name = "Repurposed Glands"
+	desc = "We shift almost all available muscle mass from the arms to the legs, disabling the former but making us unable to be downed for 15 seconds. Costs 10 chemicals."
+	helptext = "Disables your arms and retracts bioweaponry, but regenerates your legs, grants you speed, and wakes you up from any stun."
 	button_icon_state = "adrenaline"
-	chemical_cost = 30
-	dna_cost = 2
-	req_human = TRUE
-	req_stat = UNCONSCIOUS
+	chemical_cost = 10
+	dna_cost = 1
+	req_human = FALSE
+	req_stat = CONSCIOUS
 
 //Recover from stuns.
-/datum/action/changeling/adrenaline/sting_action(mob/living/user)
+/datum/action/changeling/adrenaline/sting_action(mob/living/carbon/user)
 	..()
-	to_chat(user, span_notice("Energy rushes through us."))
-	user.SetKnockdown(0)
-	user.set_resting(FALSE)
+	to_chat(user, span_changeling("Our arms feel weak, but our legs become unstoppable!"))
+
+	for(var/datum/action/changeling/weapon/weapon_ability in user.actions)
+		weapon_ability.unequip_held(user)
+
+	// Destroy legcuffs with our IMMENSE LEG STRENGTH.
+	if(user.legcuffed)
+		var/obj/O = user.get_item_by_slot(ITEM_SLOT_LEGCUFFED)
+		if(!istype(O))
+			return FALSE
+		qdel(O)
+		user.visible_message(span_warning("[user]'s legs suddenly rip [O] apart!"), \
+			span_warning("We rip apart our leg restraints!"))
+
+	// Regenerate our legs only.
+	var/our_leg_zones = (GLOB.all_body_zones - GLOB.leg_zones)
+	user.regenerate_limbs(excluded_zones = our_leg_zones) // why is this exclusive rather than inclusive
+
+	// Upgrade our stamina, lose our arms!
+	user.max_stamina *= 2
```

Comment:
Did you test this and does it work?
IIRC stamcrit is based on maxhealth and not max_stamina.

Reply:
fuckin wot

Reply:
![image](https://github.com/tgstation/tgstation/assets/53100513/496c3468-43ae-47c6-8c45-1d7bbca6784f)

??????????????????????????????? ok